tests/DCPS/MultiTopic/MultiTopicTest.cpp

```diff
@@ -7,25 +7,305 @@
+  TAO::String_Manager& misc() { return value_.misc; }
+  MiscUnion& misc_union() { return value_.misc_union; }
+
+  operator Resulting() const { return value_; }
+
+private:
+  Resulting value_;
+};
+
+#  define ENUM_WRAPPER(TYPE, MEMBER) (MEMBER)
+
+#elif defined(CPP11_MAPPING)
+template <typename T>
+bool sequences_are_not_equal(const T& a, const T& b)
+{
+  return a != b;
+}
+
+using LocationInfoWrapper = LocationInfo;
+using PlanInfoWrapper = PlanInfo;
+using MoreInfoWrapper = MoreInfo;
+using UnrelatedInfoWrapper = UnrelatedInfo;
+using ResultingWrapper = Resulting;
+
+#  define ENUM_WRAPPER(TYPE, MEMBER) TYPE::MEMBER
+
+#endif
+
+template <typename T>
+size_t get_sequence_length(T& seq)
+{
+#ifdef CLASSIC_MAPPING
+  return seq.length();
+#elif defined(CPP11_MAPPING)
+  return seq.size();
+#endif
+}
+
+template <typename T>
+void set_sequence_length(T& seq, size_t length)
+{
+#ifdef CLASSIC_MAPPING
+  seq.length(length);
+#elif defined(CPP11_MAPPING)
+  seq.resize(length);
+#endif
+}
+
+std::ostream& operator<<(std::ostream& os, FlightType value)
+{
+  switch (value) {
+  case ENUM_WRAPPER(FlightType, visual):
+    os << "visual";
+    return os;
+  case ENUM_WRAPPER(FlightType, instrument):
+    os << "instrument";
+    return os;
+  case ENUM_WRAPPER(FlightType, mixed):
+    os << "mixed";
+    return os;
+  }
+  throw std::runtime_error("Invalid value passed to operation<<(std::ostream, FlightType)");
+}
+
+bool operator!=(const MiscUnion& a, const MiscUnion& b)
+{
+  if (a._d() != b._d()) {
+    return true;
+  }
+  switch (a._d()) {
+  case 0:
+    return a.long_value() != b.long_value();
+  case 1:
+  case 2:
+#if defined(CLASSIC_MAPPING)
+    return std::strcmp(a.string_value(), b.string_value());
+#elif defined(CPP11_MAPPING)
+    return a.string_value() != b.string_value();
+#endif
+  case 3:
+    return a.char_value() != b.char_value();
+  default:
+    throw std::runtime_error("Invalid discriminator in MiscUnion");
+  }
+}
+
+std::ostream& operator<<(std::ostream& os, const MiscUnion& value)
+{
+  os << value._d() << '/';
+  switch (value._d()) {
+  case 0:
+    return os << value.long_value();
+    break;
+  case 1:
+  case 2:
+    return os << '"' << value.string_value() << '"';
+    break;
+  case 3:
+    return os << value.char_value();
+    break;
+  default:
+    os << "Invalid(" << value._d() << ")";
+    break;
+  }
+  return os;
+}
+
+template <typename T>
+bool arrays_are_not_equal(const T* a, const T* b, size_t length)
+{
+  for (size_t i = 0; i < length; i++) {
+    if (a[i] != b[i]) {
+      return true;
+    }
+  }
+  return false;
+}
+
+template <typename T>
+void print_array(std::ostream& os, const T* array, size_t length)
+{
+  for (size_t i = 0; i < length; i++) {
+    if (i) {
+      os << ", ";
+    }
+    os << array[i];
+  }
+}
+
+template <typename T>
+void print_sequence(std::ostream& os, const T& seq) {
+  for (size_t i = 0; i < get_sequence_length(seq); i++) {
```

Comment:
```suggestion
  for (unsigned int i = 0; i < get_sequence_length(seq); ++i) {
```
avoids warnings on some compilers due to narrowing from 64-bit (unsigned) size_t to 32-bit unsigned int used by sequence `operator[]`